### ISSUE TITLE
Add registration example and tutorial

### ISSUE THREAD

**Issue by dusktreader** [dusktreader/flask-praetorian#148]

**Comment by pahrohfit** [comment]
example using [flask-restful](https://github.com/flask-restful/flask-restful):

**settings.py**
```
    PRAETORIAN_HASH_SCHEME = 'pbkdf2_sha512'
    PRAETORIAN_CONFIRMATION_SENDER = 'no-reply@praetorian.sample'
    PRAETORIAN_CONFIRMATION_SUBJECT = 'Welcome to Praetorian-Sample.  Please validate your email.'
    PRAETORIAN_EMAIL_TEMPLATE = 'praetorian_sample/templates/security/registration_email.html'
```

**restful view**
```
"""
    Restful Class for user registration process.

    GET
        this performs validation of a user signup token.  this would be passed by the user
          from what they received via email (or REST response) from the POST call.
        this is responsible for performing any updates to the user to set them as 'active',
          or otherwise complete registration. 

    POST
        this creates the user record, minus some form of 'active' flag, and returns a token
          or link with the token for a GET call.
"""
class Register(Resource):
  def options(self):
  pass

  def get(self, token=None):
    """
    Finalize a new user register a new user if they supply a valid token.

    example::
       $ curl http://localhost:5000/api/v1/auth/register/<token>
    """
    if not token:
        return({'status': "failure", 'message': "You must supply a valid registration token"}, 401)

    try:
        valid_token = guard.validate_confirmation(token)
    except Exception as e:
        logging.error("Failed to validate user token for registration: {}".format(repr(e)))
        return({'status': "failure", 'message': "You must supply a valid registration token"}, 401)

    """ perform user finish action here, like flipping the active flag """
    try:
        updated_user = users.User.objects(id=valid_token['id']).first()
        updated_user.update(active=True)
    except Exception as e:
        logging.error("Failed to activate user by token registration: {}".format(repr(e)))
        return({'status': "failure", 'message': "Failed to finalize register activation.  Please try again."}, 401)

    return({'status': "success", 'message': "Successfully finished registration for {}".format(valid_token['id'])}, 200)

  def post(self):
    """
    Register a new user if they supply the correct info, sending a welcome email
      with a confirmation token/link in the body for the user to provide back to us
      via the GET Class view.

    example::
       $ curl http://localhost:5000/api/v1/auth/register -X POST \
         -d '{"first_name":"that","last_name":"guy","phone_number":"2222222222","email":"thatguy@example.com","password":"password"}'
    """

    req = request.get_json(force=True)
    logging.info("Register Info: {}".format(req))

    ## default to status check ##
    try:
        fail_validate = marsh.RegisterSchema().validate(req)
    except ValidationError as err:
        logging.error("Register Validation Failure: {}".format(err.messages))
        return "Register Validation Failure: {}".format(err.messages)

    if fail_validate:
        logging.error("Register Validation Failure: {}".format(fail_validate))
        return "Register Validation Failure: {}".format(fail_validate)

    try:
        hashed_pass = guard.hash_password(req['password'])
        user = users.User(email=req['email'], password=hashed_pass,
                          first_name=req['first_name'], last_name=req['last_name'],
                          phone_number=req['phone_number']).save()

        notify = guard.send_registration_email(user=user)
        ret = {'status': 'success', 'message': 'successfully registered user {}'.format(user)}
        return (ret, 200)

    except Exception as e:
        ret = {'status': 'failure',
                  'message': 'failed to register user',
                  'errors': repr(e)}
        return (ret, 420)
```

**Comment by pahrohfit** [comment]
Anything else, let me know

any other wishlist ideas?

**Comment by dusktreader** [comment]
@pahrohfit just fyi, I'm still working on building out the tutorials a lot. I've also made some revisions. This is all coming soon. I'm hoping to get everything wrapped up for a 1.0 release as soon as possible

**Comment by vpassapera** [comment]
I used this in conjunction in a chained user provider along with ldap, it works great.

Looking forward to a pypi release!

**Comment by dusktreader** [comment]
Addressed by PR #166 

Tutorial for flask-praetorian has been split off into it's own project: https://github.com/dusktreader/flask-praetorian-tutorial

Commits coming soon
